### PR TITLE
[Runtime][MISRA-C][Bundle] Bundle deployment with static linking

### PR DESCRIPTION
**Pull request by mehrdadhe**
Current implementation in MISRA-C bundle deployment depends on dynamic linker. This PR adds demo and test which does not rely on dynamic linking using dlopen function.

### CODE REVIEW

**Comment by tmoreau89** [comment]
@liangfu @siju-samuel

File: apps/bundle_deploy/bundle_static.c [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * under the License.
+ */
+
+#include "bundle.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include "runtime.c"
+
+
+// /*! \brief macro to do C API call */
```
Comment (tmoreau89) [inline apps/bundle_deploy/bundle_static.c]: Can we get those lines cleaned up?

File: apps/bundle_deploy/test_static.c [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  
+  tvm_runtime_get_output(handle, 0, &output);
+  gettimeofday(&t4, 0);
+
+  for (auto i = 0; i < 10 * 5; ++i) {
+    assert(fabs(output_storage[i] - result_storage[i]) < 1e-5f);
+    if (fabs(output_storage[i] - result_storage[i]) >= 1e-5f) {
+      printf("got %f, expected %f\n", output_storage[i], result_storage[i]);
+    }
+    // printf("out: %f, exp: %f\n", output_storage[i], result_storage[i]);
```
Comment (tmoreau89) [inline apps/bundle_deploy/test_static.c]: cleanup here too

File: apps/bundle_deploy/bundle_static.c (hunk 1 above)
Comment (mehrdadhe) [reply apps/bundle_deploy/bundle_static.c]: Done!

File: apps/bundle_deploy/test_static.c (hunk 2 above)
Comment (mehrdadhe) [reply apps/bundle_deploy/test_static.c]: Done!

File: apps/bundle_deploy/bundle.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                                   const char * name, 
+                                   DLTensor * tensor);
+
+TVM_DLL void tvm_runtime_run(void * runtime);
+
+TVM_DLL void tvm_runtime_get_output(void * runtime,
+                                    int32_t index, 
+                                    DLTensor * tensor);
+
+#endif /* TVM_APPS_BUNDLE_H */
```
Comment (siju-samuel) [inline apps/bundle_deploy/bundle.h]: Add a CR at last line

File: apps/bundle_deploy/bundle_static.c [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  TVMFuncGetGlobal("tvm.graph_runtime.run", (TVMFunctionHandle*)&TVMGraphRuntime_Run);
+  TVMGraphRuntime_Run(runtime);
+}
+
+TVM_DLL void tvm_runtime_get_output(void * runtime, int32_t index, DLTensor * tensor) {
+  int (*TVMGraphRuntime_GetOutput)(TVMModuleHandle, const int32_t, DLTensor *);
+  TVMFuncGetGlobal("tvm.graph_runtime.get_output", (TVMFunctionHandle*)&TVMGraphRuntime_GetOutput);
+  TVMGraphRuntime_GetOutput(runtime, index, tensor);
+}
+
```
Comment (siju-samuel) [inline apps/bundle_deploy/bundle_static.c]: Remove this extra line

File: apps/bundle_deploy/Makefile [hunk 5]
```diff
@@ -33,22 +33,36 @@ PKG_LDFLAGS = -pthread
 
 build_dir := build
 
-demo: $(build_dir)/demo $(build_dir)/bundle.so $(build_dir)/bundle_c.so $(build_dir)/cat.bin
-	TVM_NUM_THREADS=1 $(build_dir)/demo $(build_dir)/bundle.so $(build_dir)/cat.bin
-	TVM_NUM_THREADS=1 $(build_dir)/demo $(build_dir)/bundle_c.so $(build_dir)/cat.bin
+demo_dynamic: $(build_dir)/demo_dynamic $(build_dir)/bundle.so $(build_dir)/bundle_c.so $(build_dir)/cat.bin
```
Comment (siju-samuel) [inline apps/bundle_deploy/Makefile]: apps/bundle_deploy/README.md needs to be updated?

File: apps/bundle_deploy/bundle.h [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef TVM_APPS_BUNDLE_H
```
Comment (siju-samuel) [inline apps/bundle_deploy/bundle.h]: #ifndef TVM_APPS_BUNDLE_DEPLOY_BUNDLE_H_

File: apps/bundle_deploy/demo_static.c [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+  output.shape = &out_shape;
+  output.strides = NULL;
+  output.byte_offset = 0;
+
+  tvm_runtime_get_output(handle, 0, &output);
+  gettimeofday(&t4, 0);
+
+  float max_iter = -FLT_MAX;
+  int32_t max_index = -1;
+  for (auto i = 0; i < 1000; ++i) {
```
Comment (siju-samuel) [inline apps/bundle_deploy/demo_static.c]: Suggest to use a macro for output_storage len 1000,  3 places its used

Comment (mehrdadhe) [reply apps/bundle_deploy/demo_static.c]: Added, thanks for suggestion.

File: apps/bundle_deploy/Makefile (hunk 5 above)
Comment (mehrdadhe) [reply apps/bundle_deploy/Makefile]: I updated the README file. Please let me know if you have any suggestions.

**Comment by tqchen** [comment]
cc @liangfu

**Comment by liangfu** [comment]
LGTM

File: apps/bundle_deploy/Makefile (hunk 5 above)
[the remark this answers is not in the document]
Comment (mehrdadhe) [reply apps/bundle_deploy/Makefile]: Added!

**Comment by mehrdadhe** [comment]
@tmoreau89  I have pushed a commit.

**Comment by tmoreau89** [comment]
Thank you @mehrdadhe @siju-samuel @liangfu! The PR has been merged!

**Pull request merged**